code/modules/pool/pool_controller.dm

```diff
@@ -0,0 +1,358 @@
+			for(var/X in W.reagents.reagent_list)
+				var/datum/reagent/R = X
+				if(R.reagent_state == SOLID)
+					to_chat(user, "The pool cannot accept reagents in solid form!.")
+					return
+				else
+					beaker =  W
+					user.dropItemToGround(W)
+					W.forceMove(src)
+					to_chat(user, "You add the beaker to the machine!")
+					updateUsrDialog()
+					cur_reagent = "[R.name]"
+					for(var/I in linkedturfs)
+						var/turf/open/pool/P = I
+						if(P.reagents)
+							P.reagents.clear_reagents()
+							P.reagents.add_reagent(R.id, 100)
+					if(GLOB.adminlog)
+						log_game("[key_name(user)] has changed the [src] chems to [R.name]")
+						message_admins("[key_name_admin(user)] has changed the [src] chems to [R.name].")
+					timer = 15
+		else
+			to_chat(user, "<span class='notice'>This machine only accepts full large beakers of one reagent.</span>")
+			return
+	else if(panel_open && is_wire_tool(W))
+		wires.interact(user)
+	else
+		return ..()
+
+/obj/machinery/poolcontroller/screwdriver_act(mob/living/user, obj/item/W)
+	. = ..()
+	if(.)
+		return TRUE
+	cut_overlays()
+	panel_open = !panel_open
+	to_chat(user, "You [panel_open ? "open" : "close"] the maintenance panel.")
+	W.play_tool_sound(src)
+	if(panel_open)
+		add_overlay("wires")
+	return TRUE
+
+//procs
+/obj/machinery/poolcontroller/proc/shock(mob/user, prb)
+	if(stat & (BROKEN|NOPOWER))		// unpowered, no shock
+		return FALSE
+	if(!prob(prb))
+		return FALSE
+	var/datum/effect_system/spark_spread/s = new /datum/effect_system/spark_spread
+	s.set_up(5, 1, src)
+	s.start()
+	if(electrocute_mob(user, get_area(src), src, 0.7))
+		return TRUE
+	else
+		return FALSE
+
+/obj/machinery/poolcontroller/proc/poolreagent()
+	for(var/X in linkedturfs)
+		var/turf/open/pool/W = X
+		for(var/mob/living/carbon/human/swimee in W)
+			if(beaker && cur_reagent && W.reagents)
+				for(var/Q in W.reagents.reagent_list)
+					var/datum/reagent/R = Q
+					if(R.reagent_state == SOLID)
+						R.reagent_state = LIQUID
+				W.reagents.reaction(swimee, VAPOR, 0.03) //3 percent
+				for(var/Q in W.reagents.reagent_list)
+					var/datum/reagent/R = Q
+					swimee.reagents.add_reagent(R.id, 0.5) //osmosis
+		for(var/obj/objects in W)
+			if(beaker && cur_reagent && W.reagents)
+				W.reagents.reaction(objects, VAPOR, 1)
+			reagenttimer = 4
+	changecolor()
+
+
+/obj/machinery/poolcontroller/process()
+	updatePool() //Call the mob affecting proc)
+	if(timer > 0)
+		timer--
+		updateUsrDialog()
+	if(reagenttimer > 0)
+		reagenttimer--
+	if(stat & (NOPOWER|BROKEN))
+		return
+	else if(!reagenttimer && !drained)
+		poolreagent()
+
+/obj/machinery/poolcontroller/proc/updatePool()
+	if(!drained)
+		for(var/X in linkedturfs) //Check for pool-turfs linked to the controller.
+			var/turf/open/pool/W = X
+			for(var/mob/living/M in W) //Check for mobs in the linked pool-turfs.
+				switch(temperature) //Apply different effects based on what the temperature is set to.
```

Comment:
There has to be a better way to handle this. What if temperatures were defines or something? And how about making is always adjust heat by some amount into the direction of the heat?